core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -694,6 +694,22 @@ class Ruleset {
                     RulesetErrorSeverity.Warning
                 )
             }
+            for (unique in improvement.uniqueObjects) {
+                if (unique.type == UniqueType.PillageYieldRandom || unique.type == UniqueType.PillageYieldFixed) {
+                    val params = Stats.parse(unique.params[0])
+                    if (params.values.any { it < 0 }) lines.add(
+                        "${improvement.name} cannot have a negative value for a pillage yield!",
+                        RulesetErrorSeverity.Error
+                    )
+                }
+            }
+            if ((improvement.hasUnique(UniqueType.PillageYieldRandom) || improvement.hasUnique(UniqueType.PillageYieldFixed))
```

Comment:
I think I see what you're saying. I changed the `hasUnique` function call to use the `StateForConditions.IgnoreConditionals` flag.

Reply:
Conditionals can be complicated. Just learned a little about them :cat:

Reply:
Actually, the Unpillagable one wouldn't really work with a conditional anyway, but pillaging - a mod might want to forbid that , say, if below -19 happinyess, and it would work, right? But doesn't hurt.